extensions/binding/org.eclipse.smarthome.binding.weatherunderground/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,459 @@
+			</channel>
+			<channel id="minTemperature" typeId="temperature">
+				<label>Minimum Temperature</label>
+				<description>Minimum temperature</description>
+			</channel>
+			<channel id="maxTemperature" typeId="temperature">
+				<label>Maximum Temperature</label>
+				<description>Maximum temperature</description>
+			</channel>
+			<channel id="relativeHumidity" typeId="relativeHumidity" />
+			<channel id="probaPrecipitation" typeId="probaPrecipitation" />
+			<channel id="precipitationDay" typeId="rain" />
+			<channel id="snow" typeId="snow" />
+			<channel id="maxWindDirection" typeId="windDirection">
+				<label>Maximum Wind Direction</label>
+				<description>Maximum wind direction</description>
+			</channel>
+			<channel id="maxWindDirectionDegrees" typeId="windDirection-degrees">
+				<label>Maximum Wind Direction (°)</label>
+				<description>Maximum wind direction in degrees</description>
+			</channel>
+			<channel id="maxWindSpeed" typeId="windSpeed">
+				<label>Maximum Wind Speed</label>
+				<description>Maximum wind speed</description>
+			</channel>
+			<channel id="averageWindDirection" typeId="windDirection">
+				<label>Average Wind Direction</label>
+				<description>Average wind direction</description>
+			</channel>
+			<channel id="averageWindDirectionDegrees" typeId="windDirection-degrees">
+				<label>Average Wind Direction (°)</label>
+				<description>Average wind direction in degrees</description>
+			</channel>
+			<channel id="averageWindSpeed" typeId="windSpeed">
+				<label>Average Wind Speed</label>
+				<description>Average wind speed</description>
+			</channel>
+			<channel id="iconUrl" typeId="iconUrl" />
+		</channels>
+	</channel-group-type>
+
+	<channel-type id="location" advanced="true">
+		<item-type>String</item-type>
+		<label>Observation Location</label>
+		<description>Weather observation location</description>
+		<state readOnly="true" pattern="%s"></state>
+	</channel-type>
+
+	<channel-type id="stationId" advanced="true">
+		<item-type>String</item-type>
+		<label>Station Id</label>
+		<description>Weather station identifier</description>
+		<state readOnly="true" pattern="%s"></state>
+	</channel-type>
+
+	<channel-type id="time" advanced="true">
+		<item-type>DateTime</item-type>
+		<label>Observation Time</label>
+		<description>Observation date and time</description>
+		<state readOnly="true"></state>
+	</channel-type>
+
+	<channel-type id="conditions">
+		<item-type>String</item-type>
+		<label>Conditions</label>
+		<description>Weather conditions</description>
+		<state readOnly="true" pattern="%s"></state>
+	</channel-type>
+
+	<channel-type id="temperature">
+		<item-type>Number</item-type>
+		<label>Temperature</label>
+		<description>Temperature</description>
+		<category>Temperature</category>
+		<state readOnly="true" pattern="%.1f" />
+		<config-description>
+			<parameter name="SourceUnit" type="text" required="true">
+				<label>Source Temperature Unit</label>
```

Comment:
By "source", I mean I ask the binding to use as source of data the °C or °F value provided by WU.
So currently, while unit management is not yet part of ESH, the binding is sending state update using the unit defined by the source configuration.
Later, when unit management is really part of ESH, I think the idea would be to create a Quantity as state, using the source setting, but then the user could choose to display °C or °F using the conversion mechanism of the new unit management.